### PR TITLE
Simplify genericapiserver tests

### PR DESCRIPTION
**Pull request by sttts**
- create empty GenericApiServier object outside helper
- set default Config values everywhere because they are set anyway

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/33640)

### CODE REVIEW

File: pkg/genericapiserver/genericapiserver_test.go [hunk 1]
```diff
@@ -174,9 +170,10 @@ func TestNewHandlerContainer(t *testing.T) {
 // TestHandleWithAuth verifies HandleWithAuth adds the path
 // to the MuxHelper.RegisteredPaths.
 func TestHandleWithAuth(t *testing.T) {
-	server, etcdserver, _, assert := setUp(t)
+	etcdserver, _, assert := setUp(t)
 	defer etcdserver.Terminate(t)
 
+	server := GenericAPIServer{}
```
Comment (liggitt) [inline pkg/genericapiserver/genericapiserver_test.go]: `&GenericAPIServer{}`?

File: pkg/genericapiserver/genericapiserver_test.go [hunk 2]
```diff
@@ -267,10 +262,11 @@ func (authn *mockAuthenticator) AuthenticateRequest(req *http.Request) (user.Inf
 // TestInstallSwaggerAPI verifies that the swagger api is added
 // at the proper endpoint.
 func TestInstallSwaggerAPI(t *testing.T) {
-	server, etcdserver, _, assert := setUp(t)
+	etcdserver, _, assert := setUp(t)
 	defer etcdserver.Terminate(t)
 
 	mux := http.NewServeMux()
+	server := GenericAPIServer{}
```
Comment (liggitt) [inline pkg/genericapiserver/genericapiserver_test.go]: `&GenericAPIServer{}`?

**Review by liggitt — COMMENTED** [review]
a couple questions about pointers, LGTM otherwise

File: pkg/genericapiserver/genericapiserver_test.go (hunk 2 above)
Comment (sttts) [inline pkg/genericapiserver/genericapiserver_test.go]: why? It's equivalent, until you pass it to some function where you have to use `&`

Comment (liggitt) [inline pkg/genericapiserver/genericapiserver_test.go]: mostly for equivalence with previous code, and not being sure where it is passed or the side effects of the functions called on it

Comment (sttts) [inline pkg/genericapiserver/genericapiserver_test.go]: Switched it to `&` and squashed.

**Pull request merged**